types/knuddels-userapps-api/index.d.ts

```diff
@@ -1,25 +1,35 @@
-// Type definitions for Knuddels UserApps API 1.00108823
+// Type definitions for Knuddels UserApps API 1.00109481
 // Project: https://developer.knuddels.de
 // Definitions by: Knuddels GmbH & Co. KG <https://github.com/Knuddels>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-// JSON definition
-export interface Json {
-	[x: string]: string | number | boolean | Date | Json | JsonArray;
-}
-export interface JsonArray extends Array<string | number | boolean | Date | Json | JsonArray> { }
-
-// serializable objects (for persistence)
+// helper types
+export type JsonData = string | number | boolean | Date | Json | JsonArray;
+export type KnuddelsJsonData = string | number | boolean | Date | KnuddelsJson | KnuddelsJsonArray | KnuddelsSerializable;
 export type KnuddelsSerializable = string | number | boolean | User | BotUser;
-export interface KnuddelsJson {
-	[x: string]: string | number | boolean | Date | KnuddelsJson | KnuddelsJsonArray | KnuddelsSerializable;
-}
-export interface KnuddelsJsonArray extends Array<string | number | boolean | Date | KnuddelsJson | KnuddelsJsonArray | KnuddelsSerializable> { }
-
-// "data" that may be send between apps and between server and client
 export type KnuddelsEvent = string | Json | KnuddelsEventArray;
-export interface KnuddelsEventArray extends Array<string | KnuddelsEvent | KnuddelsEventArray> { }
 
+// helper interfaces
+declare global {
```

Comment:
Is there a reason you have to add these to the global scope?

Reply:
These interfaces/types have been on the global scope already, before the last PR. In the last PR a few weeks ago I accidentally removed them from the scope. This was caused by re-writing our algorithm that creates this file. We do not maintain it manually, it gets generated with information from our documentation. And since the developers using our API need to use these interfaces for some of our APIs we have to make them available. Moving them inside a namespace for example would be a breaking change we don't want to introduce now.